idaes/core/util/env_info.py

```diff
@@ -0,0 +1,130 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+
+__Author__ = "John Eslick"
+
+import sys
+import platform
+from collections import OrderedDict
+import pkg_resources
+
+import idaes
+import pyomo
+import pyomo.environ as pyo
+
+import idaes.ver as ver
+
+class EnvironmentInfo():
+    """Get information about IDAES and the environment IDAES is running in,
+    including OS and Critial dependency versions."""
+
+    known_solvers = [
+        "ipopt",
+        "ipopt_sens",
+        "ipopt_l1",
+        "bonmin",
+        "couenne",
+        "cbc",
+        "k_aug",
+        "dot_sens",
+    ]
+
+    def __init__(self, addtional_solvers=()):
+        # Get idaes version from ver module.  This works even if you just
+        # check a new version our from github, have IDAES installed in-place
+        # and don't reinstall, which is likely mode for a lot of developers
+        self.git_hash = ver.gh
+        self.package_version = ver.package_version
+        self.version_string = ver.__version__
+        self.bin_directory = idaes.bin_directory
+        self.data_directory = idaes.data_directory
+        self.global_config = idaes._global_config_file
+        self.local_config = idaes._local_config_file
+        # Get Python info
+        self.python_version = sys.version
```

Comment:
It could be useful to get the path to the Python executable (from `sys.executable`) here as well.